### ISSUE TITLE
Git Commands:

### ISSUE THREAD

**Issue by esterlev23** [skyvar/self-learning#1]
I will provide explanations based on a Google search

**Comment by bdsoha** [comment]
- Who are you working with? 
- Where is the description?
- What is the issue's title?
- I don't want someone else's answers *(Google search)*, I want your answers. You are meant to actually use the commands in advance use cases to explain when they are relevant etc.

**Comment by esterlev23** [comment]
**I'm working with @MiriamRotenshtein .**
**description**: Of course, the explanation will be mine, but there are commands I don't know, so I will look for information about them on Google.
**Issue's title**: Git Commands

**Comment by bdsoha** [comment]
> **I'm working with @MiriamRotenshtein .** **description**: Of course, the explanation will be mine, but there are commands I don't know, so I will look for information about them on Google. **Issue's title**: Git Commands

The **description** is at the top of the page, change it [here](https://github.com/skyvar/self-learning/issues/1#issue-1631691987).

**Comment by MiriamRotenshtein** [comment]
> provide details of:
Use-cases (The topics should include advances uses).
Pros/cons.
Coding examples.
Alternatives.
Screenshots.

Do we need to provide for each command?

**Comment by bdsoha** [comment]
Yes, all the mentioned commands.

**Comment by bdsoha** [comment]
@MiriamRotenshtein Please change your profile name to use title-case *(i.e. `Miriam Rotenshtein `)*, as well as your `@skyvar.co.il` email.

**Comment by bdsoha** [comment]
Any questions? Updates?

**Comment by esterlev23** [comment]
In the meantime we are exploring the commands.
We noticed that there are commands that it is impossible to provide all the sections,
For example: an alternative to `git remote` etc.

**Comment by bdsoha** [comment]
Good comment 👍. 
You are only expected to provide alternatives *when* available.
There are many cases that do not have an alternative.

**Comment by bdsoha** [comment]
@esterlev23 Please also add a short description and use-case for `.gitconfig` and `.gitattributes`.

**Comment by esterlev23** [comment]
We are now writing file that deals with the pre-commitment issue.
We have seen that this is a somewhat complex issue and full of details.
We wanted to ask:
Is it correct to give links on the md file to the link of https://pre-commit.com
For example:
First of all [install pre-commit](https://pre-commit.com/#install)
???

**Comment by bdsoha** [comment]
I don't think that this needs to be a *tutorial* or *user manual* on how to install and use the tool.
It should be more along the lines of an *overview* so that future users will know **when** and **why** the tool is needed.

You can provide 1 or 2 code snippets or screenshots *(if needed)* to present the overall use-case.

Regarding links, absolutely include links to sources for future reference.

**Comment by bdsoha** [comment]
In the future, development and research should be done iteratively. 
Instead of upload a *batch* at once.

That way, you receive feedback and insight immediately.

**Comment by esterlev23** [comment]
Why we can't doing PR?

**Comment by bdsoha** [comment]
> Why we can't doing PR?

Figure it out.
